aws/table_aws_iot_greengrass_deployment.go

```diff
@@ -0,0 +1,226 @@
+				Hydrate:     getIotGreengrassDeployment,
+			},
+			{
+				Name:        "deployment_policies",
+				Description: "The deployment policies for the deployment. These policies define how the deployment updates components and handles failure.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getIotGreengrassDeployment,
+			},
+			{
+				Name:        "iot_job_configuration",
+				Description: " The job configuration for the deployment configuration. The job configuration specifies the rollout, timeout, and stop configurations for the deployment configuration.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getIotGreengrassDeployment,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DeploymentName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getIotGreengrassDeployment,
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listIotGreengrassDeployments(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := IoTGreengrassClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_iot_greengrass_deployment.listIotGreengrassDeployments", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	input := &greengrassv2.ListDeploymentsInput{
+		MaxResults: aws.Int32(maxLimit),
+	}
+	if d.EqualsQualString("target_arn") != "" {
+		input.TargetArn = aws.String(d.EqualsQualString("target_arn"))
+	}
+	if d.EqualsQualString("parent_target_arn") != "" {
+		input.ParentTargetArn = aws.String(d.EqualsQualString("parent_target_arn"))
+	}
+
+	paginator := greengrassv2.NewListDeploymentsPaginator(svc, input, func(o *greengrassv2.ListDeploymentsPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		// apply rate limiting
+		d.WaitForListRateLimit(ctx)
+
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_iot_greengrass_deployment.listIotGreengrassDeployments", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.Deployments {
+			d.StreamListItem(ctx, item)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getIotGreengrassDeployment(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
@ParthaI Please update the function name from `getIotGreengrassDeployment` to `getIoTGreengrassDeployment`